### PR TITLE
Added unit tests for Switch component

### PR DESCRIPTION
**Pull request by neborn**
Wrote API tests for for the Switch component.

### CODE REVIEW

File: test/switch/switch.html [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+  <script>
+
+    Inject.setExpires( 0 );
+    Inject.clearCache();
+    Inject.setModuleRoot( 'http://localhost:1337/' );
+
+    window.LU_DEBUG = 5;
+
+    Lu.map( $( '[data-lu~=Switch]' ), 'Switch', function(){} );
```
Comment (Iheartweb) [inline test/switch/switch.html:35]: This is done for you in mappers.js - Abstract is the only one not done automatically.

**Comment by Iheartweb** [comment]
Good stuff!

Add a few event tests to emulate the state event. The easiest way to do this is to add a bunch of buttons in the switch and `$button.click`

``` html
  <div data-lu="Switch" class="lu-state-red">
    <button data-lu="Button:State" data-lu-config="{
      'Button:State': {
        states: 'blue'
      }
    }">Blue</button>
    <button data-lu="Button:State" data-lu-config="{
      'Button:State': {
        states: 'red'
      }
    }">Red</button>
    <button data-lu="Button:State" data-lu-config="{
      'Button:State': {
        states: 'red, blue'
      }
    }">Blue/Red</button>
    <button data-lu="Button:State" data-lu-config="{
      'Button:State': {
        states: 'tall'
      }
    }">Tall</button>
    <button data-lu="Button:State" data-lu-config="{
      'Button:State': {
        states: 'wide'
      }
    }">Wide</button>
    <button data-lu="Button:State" data-lu-config="{
      'Button:State': {
        states: 'tall',
        method: 'add'
      }
    }">Tall Additive</button>
    <button data-lu="Button:State" data-lu-config="{
      'Button:State': {
        states: 'wide',
        method: 'add'
      }
    }">Wide Additive</button>
    <button data-lu="Button:State" data-lu-config="{
      'Button:State': {
        states: 'tall',
        method: 'remove'
      }
    }">Tall Remove</button>
    <button data-lu="Button:State" data-lu-config="{
      'Button:State': {
        states: 'wide',
        method: 'remove'
      }
    }">Wide Remove</button>
    <button data-lu="Button:State" data-lu-config="{
      'Button:State': {
        method: 'clear'
      }
    }">Clear</button>
    <button data-lu="Button:State" data-lu-config="{
      'Button:State': {
        method: 'reset'
      }
    }">Reset</button>
  </div>
```

**Comment by Iheartweb** [comment]
Imma gonna merge this now and update the bug.

**Comment by Iheartweb** [comment]
see https://github.com/linkedin/Lu/issues/240

**Comment by neborn** [comment]
Thanks Robert. I will get that done. Looks like you've written tests for quite a few other components. Let me know what other components you would like me to start writing tests for. Thanks!

**Pull request merged**